set.mm

```diff
@@ -147402,6 +147402,42 @@ computer programs (as last() or lastChar()), the terminology used for
     JZANUBJZUQUCZUTQTURUOVAVBUOUPUQUDURUEUPUOVBUQAUFUGUHUQUOVEUPUQVCVDUIUJNABCU
     MUKUL $.
 
+  ${
+    $d l s $.
+    $( The value of a prefix operation for a length argument not in the range
+       of the word length is the empty set.  (This is due to our definition of
+       function values for out-of-domain arguments, see ~ ndmfv ).
+       (Contributed by AV, 30-Nov-2022.) $)
+    pfxnd0 $p |- ( ( W e. Word V /\ L e/ ( 0 ... ( # ` W ) ) )
+                   -> ( W prefix L ) = (/) ) $=
+      ( vs vl wcel cc0 co cpfx cn0 wn wbr w3o wb a1i wi wo com12 cvv wa cfv cfz
+      cword chash wnel wceq cle w3a df-nel elfz2nn0 notbid 3ianor 3bitrd 3orrot
+      3orass lencl pm2.24d olc ianor sylibr cop csubstr df-pfx mpt2ndm0 syl a1d
+      c0 cv notnotb clt cr nn0re ltnle syl2an pfxnd 3expia sylbird expcom com23
+      sylbir imp jaoi3 orcoms jaoi sylbi sylbid ) CBUCFZAGCUDUAZUBHZUEZCAIHVGUF
+      ZWGWJAJFZKZWHJFZKZAWHUGLZKZMZWKWGWJAWIFZKZWLWNWPUHZKZWRWJWTNWGAWIUIOWGWSX
+      AWSXANWGAWHUJOUKXBWRNWGWLWNWPULOUMWRWGWKWRWOWQWMMZWGWKPZWMWOWQUNXCWOWQWMQ
+      ZQXDWOWQWMUOWOXDXEWGWOWKWGWNWKBCUPZUQRWMWQXDWMXDWQWMWKWGWMCSFZWLTKZWKWMXG
+      KZWMQXHWMXIURXGWLUSUTDEDVHGEVHVAVBHICASJDEVCVDVEVFWMKZWQXDXJWLWQXDPWLVIWL
+      WGWQWKWGWLWQWKPWGWLTWQWHAVJLZWKWGWHVKFZAVKFXKWQNWLWGWNXLXFWHVLVEAVLWHAVMV
+      NWGWLXKWKABCVOVPVQVRVSVTWAWBWCWDWEWERWFWA $.
+  $}
+
+  $( A prefix is a word over the symbols it consists of.  (Contributed by AV,
+     30-Nov-2022.) $)
+  pfxwrdsymb $p |- ( S e. Word A
+                     -> ( S prefix L ) e. Word ( S " ( 0 ..^ L ) ) ) $=
```

Comment:
Unfortunately, there are no such theorems. You think of something like:
* ( S e. Word A -> ( S substring <. M , N >. ) e. Word ( S " ( M ..^ N ) ) )
* ( S e. Word A -> S e. Word ( S " ( 0 ..^ ( # ` S ) ) ) )

If you want, I can provide/proof these two theorems (~pfxwrdsymb would be a special case of the first one)

Reply:
Yes, it would seem natural that if the version for prefixes is there, it should be proved as a special case of the version of subwords. And the version for words should also be there. So it would be nice in a future PR.